services/reuse/reuse-compliance.service.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { renderReuseBadge } = require('./reuse-compliance-helper')
+
+const responseSchema = Joi.object({
+  badge: Joi.string(),
+  hash: Joi.string(),
+  last_access: Joi.date(),
+  lint_code: Joi.number(),
+  lint_output: Joi.string(),
+  status: Joi.string().required(),
```

Comment:
If we can only deal with 2 values ('compliant' and 'non-compliant') we should validate that status is exactly one of those two values, not just any old string.
Also it looks like the only value we are actually using here is status. We don't need to validate all this other stuff. i.e: if we only need the status value and we get a response back with a valid status but no `url` key we shouldn't fail to render a badge because of it.

Reply:
You can get rid of `badge`, `hash`, `last_access`, `lint_code` and `lint_output` here too. None of them are used by the badge. We only want to validate that values we need to render the badge

Reply:
FWIW, there are [4 statuses](https://git.fsfe.org/reuse/api/src/branch/master/reuse_api/badges): compliant, non-compliant, checking and unregistered

Reply:
Any (other) valid values should also be

* validated in the schema and
* added to `COLOR_MAP`